rskj-core/src/main/java/co/rsk/db/PruneService.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * This file is part of RskJ
+ * Copyright (C) 2018 RSK Labs Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package co.rsk.db;
+
+import co.rsk.config.RskSystemProperties;
+import co.rsk.core.RskAddress;
+import co.rsk.trie.TrieCopier;
+import co.rsk.trie.TrieStore;
+import co.rsk.trie.TrieStoreImpl;
+import org.ethereum.core.Blockchain;
+import org.ethereum.datasource.KeyValueDataSource;
+import org.ethereum.util.FileUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.ethereum.datasource.DataSourcePool.levelDbByName;
+import static org.ethereum.datasource.DataSourcePool.closeDataSource;
+
+/**
+ * Created by ajlopez on 21/03/2018.
+ */
+public class PruneService {
+    private static final Logger logger = LoggerFactory.getLogger("prune");
+
+    private final RskSystemProperties rskConfiguration;
+    private final PruneConfiguration pruneConfiguration;
+    private final Blockchain blockchain;
+    private final RskAddress contractAddress;
+
+    private boolean stopped;
+    private long nextBlockNumber;
+
+    public PruneService(PruneConfiguration pruneConfiguration, RskSystemProperties rskConfiguration, Blockchain blockchain, RskAddress contractAddress) {
+        this.pruneConfiguration = pruneConfiguration;
+        this.rskConfiguration = rskConfiguration;
+        this.blockchain = blockchain;
+        this.contractAddress = contractAddress;
+        this.nextBlockNumber = pruneConfiguration.getNoBlocksToWait();
+    }
+
+    public void start() {
+        this.stopped = false;
+        new Thread(this::run).start();
+        logger.info("launched");
+    }
+
+    public void run() {
+        while (this.stopped == false) {
+            long bestBlockNumber = this.blockchain.getStatus().getBestBlockNumber();
+
+            if (bestBlockNumber > nextBlockNumber) {

```

Comment:
If I read this right, this check will make prune run the node launches. Shouldn't this run every noBlocksToWait?